packages/docusaurus-theme-classic/src/theme/BlogListPage/StructuredData/index.tsx

```diff
@@ -0,0 +1,79 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import React from 'react';
+import {useBaseUrlUtils} from '@docusaurus/useBaseUrl';
+import useDocusaurusContext from '@docusaurus/useDocusaurusContext';
+import {
+  makeImageStructuredData,
+  makePersonStructuredData,
+} from '@docusaurus/theme-common';
+import type {Props} from '@theme/BlogListPage/StructuredData';
+import StructuredData from '@theme/StructuredData';
+import type {Blog, WithContext} from 'schema-dts';
+
+export default function BlogListPageStructuredData(props: Props): JSX.Element {
```

Comment:
Can all this complexity be moved to `theme-common`?

This is not related to styling and is only technical so I'd rather put a minimal amount of code here. If we introduce a tailwind theme someday we shouldn't have to copy/paste this.

Code should be something such as:


```tsx
const structuredData = useBlogListPageStructuredData(props);
return <StructuredData structuredData={blogStructuredData} />
```

Or even just:

```tsx
<BlogListPageStructuredData (...props}/>
```

Users can still swizzle this and override the logic if they want. 

Of course they can't tweak the logic we use to generate that structured data but they could still copy/paste it and create their own blog structured data utils if they really want to.

Reply:
yup - done